src/api/cars/cars.controller.ts

```diff
@@ -1,20 +1,126 @@
 import { Request, Response } from 'express'
+import { AuthRequest } from '../../auth/auth.types';
+import { User } from '../user/user.types'
 
 import { 
   getAllCars,
-  createCar
+  createCar, 
+  getCarByDriverId, 
+  getCarById,
+  deleteCar, 
+  updateCar,
+
 } from './cars.service'
 
 export const getAllCarsHandler = async (_: Request, res: Response) => {
-  const cars = await getAllCars()
+ try {
+    const cars = await getAllCars()
+
+    return res.status(200).json({ message: "Cars have been found successfully" , cars})
+
+  } catch ({ message}: any ) {
+    res.status(400).json({ message: "Cars not found" });
+  } 
+}
+
+export async function getCarHandler(req: Request, res: Response) {
+  try {
+    const { id } = req.body;
+
+    const car = await getCarById(id);
+
+    if (!car){
+      return res.status(404).json({
+        message: 'Car not found',
+      })
+    }
+
+    const data = {
+      id: car.id,
+      car_name: car.car_name,
+      type: car.type,
+      img: car.img,
+      seats: car.seats,
+      luggage: car.luggage,
+      air_conditioner: car.air_conditioner,
+      transmition: car.transmition,
+      fare_km: car.fare_km,
+      driver_id: car.driver_id,
+      created_at: car.created_at,
+      updated_at: car.updated_at
+    }
 
-  return res.status(200).json(cars)
+    return res.json({message: "Car have been found successfully", car});
```

Comment:
I consider return is not necessary for sending a response, just send the response and remember add it an status, i.e. something like:
`res.status(201).json({message: "Car have been found successfully", car});`